examples/src/main/scala/caliban/akkahttp/AuthExampleApp.scala

```diff
@@ -0,0 +1,71 @@
+package caliban.akkahttp
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{ HttpResponse, StatusCodes }
+import akka.http.scaladsl.server.Directives.{ getFromResource, path, _ }
+import akka.http.scaladsl.server.RequestContext
+import caliban.AkkaHttpAdapter.ContextWrapper
+import caliban.GraphQL._
+import caliban.RootResolver
+import caliban.interop.circe.AkkaHttpCirceAdapter
+import caliban.schema.GenericSchema
+import zio.internal.Platform
+import zio.{ FiberRef, Has, RIO, Runtime, URIO, ZIO, ZLayer }
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.io.StdIn
+
+object AuthExampleApp extends App with AkkaHttpCirceAdapter {
+
+  case class AuthToken(value: String)
+
+  type Auth = Has[FiberRef[Option[AuthToken]]]
+
+  object AuthWrapper extends ContextWrapper[Auth, HttpResponse] {
+    override def apply[R <: Auth](ctx: RequestContext)(effect: URIO[R, HttpResponse]): URIO[R, HttpResponse] =
+      ctx.request.headers.collectFirst {
+        case header if header.name.toLowerCase == "token" => header.value
+      } match {
+        case Some(token) =>
+          for {
+            _      <- ZIO.accessM[Auth](_.get.set(Some(AuthToken(token))))
+            result <- effect
+          } yield result
+        case _ =>
+          URIO.fromFunction(_ => HttpResponse(StatusCodes.Forbidden))
```

Comment:
Why not just `URIO.succeed`?

Reply:
We must retain the environment in the return type, and thus succeed will not satisfy.

Reply:
If you use `succeed`, it returns a `ZIO` with `R` = `Any`, which is valid because R is contravariant.